pkgs/by-name/co/copywrite/package.nix

```
@@ -0,0 +1,57 @@
+{
+  lib,
+  stdenv,
+  buildGoModule,
+  fetchFromGitHub,
+  installShellFiles,
+  testers,
+  copywrite,
+}:
+
+buildGoModule rec {
+  pname = "copywrite";
+  version = "0.19.0";
+
+  src = fetchFromGitHub {
+    owner = "hashicorp";
+    repo = "copywrite";
+    rev = "6ed520a710166c6094098b786c63f212604654a4"; # matches tag release
```

```suggestion
    rev = "v${version}";
```

You can checkout a tagged revision like this, no need to fiddle with the hashes.

---

I would also much prefer to use the tagged revision, but without it I can't set the `shortRev`.
Initially I approached this differently, using the tagged revision, but discussing it in the official nix discord, this was the most preferred solution.

---

Given that a git tag is actually more precise in identifying a commit than a shortened hash, I fail to see the point, but I would of course not disagree with the "official nix discord".

---

Sorry, it was a typo, I meant unofficial nix discord, mb xD
I want to match the way the original binary specifies its version (`passthru.tests.version` tests exactly that).
I can modify the PR to use the tagged revision and use `leaveDotGit` to grab the short rev in the `postUnpack` phase. I previously left that on hold due to https://github.com/NixOS/nixpkgs/issues/8567.
Would that approach be satisfactory?

---

I worked on this a bit, but wasn't able to get `leaveDotGit` to be deterministic, the hash kept changing.
The `-X github.com/hashicorp/copywrite/cmd.commit` is not super important, is just metadata the `copywriter --version` command outputs.
Made some changes to use the tagged revision and the commit hash is set manually.